caf-logging-tomcat-juli/pom.xml

```diff
@@ -0,0 +1,221 @@
+                            <shadeSourcesContent>true</shadeSourcesContent>
+                            <artifactSet>
+                                <includes>
+                                    <include>org.apache.tomcat:tomcat-juli</include>
+                                    <include>org.slf4j:jcl-over-slf4j</include>
+                                    <include>org.slf4j:slf4j-api</include>
+                                    <include>ch.qos.logback:logback-classic</include>
+                                    <include>ch.qos.logback:logback-core</include>
+                                    <include>com.fasterxml.jackson.core:jackson-core</include>
+                                    <include>org.apache.commons:commons-text</include>
+                                    <include>com.github.cafapi.logging:caf-logging-logback</include>
+                                    <include>com.github.cafapi.util:util-process-identifier</include>
+                                    <include>com.github.cafapi.logging:caf-logging-common</include>
+                                </includes>
+                            </artifactSet>
+
+                            <filters>
+                                <filter>
+                                    <artifact>org.apache.tomcat:tomcat-juli</artifact>
+                                    <includes>
+                                        <include>org/apache/juli/ClassLoaderLogManager**</include>
+                                        <include>org/apache/juli/WebappProperties.class</include>
+                                        <include>META-INF/LICENSE</include>
+                                        <include>META-INF/NOTICE</include>
+                                    </includes>
+                                </filter>
+
+                                <!-- Exclude services from jcl-over-slf4j as not needed in this classloader -->
+                                <filter>
+                                    <artifact>org.slf4j:jcl-over-slf4j</artifact>
+                                    <excludes>
+                                        <exclude>META-INF/services/**</exclude>
+                                    </excludes>
+                                </filter>
+
+                                <!--
+                                    Exclude 'ServletContainerInitializer' service from logback-classic as not needed in this classloader
+                                -->
+                                <filter>
+                                    <artifact>ch.qos.logback:logback-classic</artifact>
+                                    <excludes>
+                                        <exclude>META-INF/services/javax.servlet.ServletContainerInitializer</exclude>
+                                    </excludes>
+                                </filter>
+                            </filters>
+
+                            <relocations>
```

Comment:
This all seems very complicated.  How do we maintain this?  Might this require changes when we move to `9.0.30` and beyond?  What is the procedure for knowing how and when to update these relocations and filters and so on?

Reply:
This pom file is created based on the file https://github.com/tomcat-slf4j-logback/tomcat9-slf4j-logback/blob/tomcat9-slf4j-logback-9.0.26/pom.xml
If version changes, the corresponding released version pom file from tomcat9-slf4j-logback repo changes for relocations should be included here.